spring-data-relational/src/main/java/org/springframework/data/relational/core/dialect/SQLiteDialect.java

```diff
@@ -0,0 +1,96 @@
+package org.springframework.data.relational.core.dialect;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.data.convert.WritingConverter;
+import org.springframework.data.relational.core.sql.LockOptions;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * SQL Dialect for SQLite.
+ *
+ * @author Rudolf Schmidt
+ * @since 3.3.0
+ */
+public class SQLiteDialect extends AbstractDialect {
+
+	public static final SQLiteDialect INSTANCE = new SQLiteDialect();
+
+	private SQLiteDialect() {
+	}
+
+	@Override
+	public LimitClause limit() {
+		return new LimitClause() {
+			@Override
+			public String getLimit(final long limit) {
+				return String.format("limit %d", limit);
+			}
+
+			@Override
+			public String getOffset(final long offset) {
+				throw new UnsupportedOperationException("offset alone not supported");
+			}
+
+			@Override
+			public String getLimitOffset(final long limit, final long offset) {
+				return String.format("limit %d offset %d", limit, offset);
+			}
+
+			@Override
+			public Position getClausePosition() {
+				return Position.AFTER_ORDER_BY;
+			}
+		};
+	}
+
+	@Override
+	public LockClause lock() {
+		return new LockClause() {
+			@Override
+			public String getLock(final LockOptions lockOptions) {
+				return "with lock";
+			}
+
+			@Override
+			public Position getClausePosition() {
+				return Position.AFTER_ORDER_BY;
+			}
+		};
+	}
+
+	@Override
+	public Collection<Object> getConverters() {
+		final var converters = new ArrayList<>(super.getConverters());
```

Comment:
We do not use `final` for variable declarations and we do not use `var` for type inference.